src/components/button-minor/button-minor.test.js

```diff
@@ -0,0 +1,461 @@
+          </ButtonMinor>
+        );
+
+        buttonSubtextPreview().should("have.text", subtext);
+      }
+    );
+
+    it.each(testData)(
+      "should render Button Minor name using %s special characters",
+      (name) => {
+        CypressMountWithProviders(<ButtonMinor name={name}>Foo</ButtonMinor>);
+
+        buttonDataComponent().should("have.attr", "name", name);
+      }
+    );
+
+    it.each(testData)(
+      "should render Button Minor id using %s special characters",
+      (id) => {
+        CypressMountWithProviders(<ButtonMinor id={id}>Foo</ButtonMinor>);
+
+        buttonDataComponent().should("have.attr", "id", id);
+      }
+    );
+
+    it.each(testData)(
+      "should render tooltip message with %s special characters",
+      (tooltipMessage) => {
+        CypressMountWithProviders(
+          <ButtonMinor
+            iconType="bin"
+            iconTooltipMessage={tooltipMessage}
+            m="100px"
+          />
+        );
+
+        buttonDataComponent().children().last().realHover();
+        tooltipPreview().should("have.text", tooltipMessage);
+        cyRoot().realHover({ position: "topLeft" });
+      }
+    );
+
+    it.each([
+      [BUTTON_SIZES[0], 32],
+      [BUTTON_SIZES[1], 40],
+      [BUTTON_SIZES[2], 48],
+    ])("should render Button Minor in %s size", (size, minHeight) => {
+      CypressMountWithProviders(<ButtonMinor size={size}>{size}</ButtonMinor>);
+
+      buttonDataComponent().should("have.css", "min-height", `${minHeight}px`);
+    });
+
+    it.each([
+      [BUTTON_ICON_POSITIONS[0], "right"],
+      [BUTTON_ICON_POSITIONS[1], "left"],
+    ])(
+      "should set position to %s for icon in a button",
+      (iconPosition, margin) => {
+        CypressMountWithProviders(
+          <ButtonMinor iconType="add" iconPosition={iconPosition}>
+            IconPosition
+          </ButtonMinor>
+        );
+
+        icon().should("have.css", `margin-${margin}`, "8px");
+      }
+    );
+
+    it("should render Button Minor with full width", () => {
+      CypressMountWithProviders(<ButtonMinor fullWidth>Foo</ButtonMinor>);
+
+      buttonDataComponent().then(($el) => {
+        useJQueryCssValueAndAssert($el, "width", 1365);
+      });
+    });
+
+    it("should render Button Minor with href", () => {
+      CypressMountWithProviders(
+        <ButtonMinor href="https://carbon.sage.com/">Foo</ButtonMinor>
+      );
+
+      buttonDataComponent().should(
+        "have.attr",
+        "href",
+        "https://carbon.sage.com/"
+      );
+    });
+
+    it.each([
+      ["true", "white-space"],
+      ["false", "flex-wrap"],
+    ])(
+      "should render the Button Minor text without wrapping when noWrap is called",
+      (booleanState, cssValue) => {
+        CypressMountWithProviders(
+          <ButtonMinor noWrap={booleanState}>
+            {" "}
+            Long long long long long text{" "}
+          </ButtonMinor>
+        );
+
+        if (booleanState === true) {
```

Comment:
Is this correct? This is not testing that if `noWrap` is true it will have just one ccs value or not. It is testing that if `noWrap` is true then `white-space` will be a value or if `noWrap` is false then `flex-wrap` will be a value.